tapdb/assets_store.go

```diff
@@ -2401,20 +2401,35 @@ func insertAssetTransferOutput(ctx context.Context, q ActiveAssetsStore,
 		return fmt.Errorf("unable to insert script key: %w", err)
 	}
 
+	// Mark the output proof as undelivered if it is intended for a
+	// counterpart. A proof is considered destined for a remote peer if a
+	// proof courier address is present or if the script key is not local.
+	//
+	// If the transfer output proof is not intended for a counterpart the
+	// `proofDeliverComplete` field will be left as NULL.
+	var proofDeliverComplete sql.NullBool
+	if len(output.ProofCourierAddr) > 0 || !output.ScriptKeyLocal {
```

Comment:
Here we need the conditions under which a proof will be delivered. If a proof is to be delivered then `proofDeliverComplete` is set to `false` at this point. Which means: "should be delivered, but hasn't yet". Otherwise, if there is no proof to be delivered then we leave it as `null`.

So maybe we only need `if len(output.ProofCourierAddr) > 0` here?

Reply:
I've added more to the comment to make that clear.

Reply:
I'm not sure we can just rely on the proof delivery address being set. It's true that we never set the delivery address for tombstone outputs or burns. But an RPC user might, mistakenly.
So I really think we should extract all the conditions linked above into a method on the `TransferOutput` struct, something like `ShouldDeliverProof`. That also makes the `ChainPorter.transferReceiverProof` way slimmer.